server/db/driver/pg/orders.go

```diff
@@ -1009,6 +959,119 @@ func (a *Archiver) UserOrders(ctx context.Context, aid account.AccountID, base,
 	return orders, statuses, err
 }
 
+// UserOrderStatuses retrieves the statuses and filled amounts of the orders
+// with the provided order IDs for the given account in the market specified
+// by a base and quote asset.
+// The number and ordering of the returned statuses is not necessarily the same
+// as the number and ordering of the provided order IDs. It is not an error if
+// any or all of the provided order IDs cannot be found for the given account
+// in the specified market.
+func (a *Archiver) UserOrderStatuses(aid account.AccountID, base, quote uint32, oids []order.OrderID) ([]*db.OrderStatus, error) {
+	marketSchema, err := a.marketSchema(base, quote)
+	if err != nil {
+		return nil, err
+	}
+
+	// Active orders.
+	fullTable := fullOrderTableName(a.dbName, marketSchema, true)
+	activeOrderStatuses, err := a.userOrderStatusesFromTable(fullTable, aid, oids)
```

Comment:
So we reject requests exceeding this cap? Or we query the db in batches?

Reply:
Seems like the query can handle a lot though. I modified the `orders_online_test` to search for millions of oids and only got a out of memory error at 5 million oids. The oids did not exist in the db however, don't know if that makes a difference.

I'm retesting with having thousands/millions of orders in the db and running this query with all the oids and see where there's a bottleneck.

Reply:
I modified the `orders_online_test : TestUserOrderStatuses` to save 100,002 records to db. The tests run 3 queries, the first expects 100,000 matches, the second returns 2 matches and the third returns no matches. All worked as expected without issues. Looks like if we're adding a limit to this route, it'll be just to prevent really excessive calls.

Reply:
k, how about 10k then?  If many people do it at once it could be an attack vector.